queries/cylc/highlights.scm

```diff
@@ -0,0 +1,104 @@
+(include_statement
+  directive: _ @keyword.directive
+  path: _ @string.special.path)
+
+(comment) @comment
+
+(graph_section
+  brackets_open: _ @punctuation.bracket
+  name: _? @label
+  brackets_close: _ @punctuation.bracket)
+
+(task_section
+  brackets_open: _ @punctuation.bracket
+  name: (_
+    (task_name) @function)
+  brackets_close: _ @punctuation.bracket)
+
+(top_section
+  brackets_open: _ @punctuation.bracket
+  name: _? @label
+  brackets_close: _ @punctuation.bracket)
+
+(sub_section_1
+  brackets_open: _ @punctuation.bracket
+  name: _? @label
+  brackets_close: _ @punctuation.bracket)
+
+(sub_section_2
+  brackets_open: _ @punctuation.bracket
+  name: _? @label
+  brackets_close: _ @punctuation.bracket)
+
+(runtime_section
+  brackets_open: _ @punctuation.bracket
+  name: _? @label
+  brackets_close: _ @punctuation.bracket)
+
+(graph_setting
+  key: (_) @number
+  operator: (_)? @operator)
+
+(quoted_graph_string
+  quotes_open: _ @string
+  quotes_close: _ @string)
+
+(multiline_graph_string
+  quotes_open: _ @string
+  quotes_close: _ @string)
+
+[
+  (graph_logical)
+  (graph_arrow)
+  (graph_parenthesis)
+] @keyword.operator
+
+(intercycle_annotation
+  (recurrence) @number)
+
+(graph_task
+  xtrigger: _? @keyword.modifier
+  suicide: _? @keyword.modifier
+  name: _ @function)
+
+(task_parameter
+  "<" @punctuation.bracket
+  name: (_)? @keyword.directive
+  ","? @punctuation.delimiter
+  "="? @operator
+  selection: (_)? @keyword.directive
+  ">" @punctuation.bracket)
+
+(intercycle_annotation
+  "[" @punctuation.bracket
```

Comment:
These should just be captured in a simple list, not inside a complicated pattern.

(And in general your grammar overuses fields; if you want to capture `xtrigger: _`, then it would be better to name the node instead and capture `(xtrigger)` itself -- then you don't need all these complicated patterns that are just variations on a theme.)

Reply:
> And in general your grammar overuses fields.

I'll make that a priority for the next version of the grammar. Sadly Tree-sitter lacks good resources for "style" and best practices so I'm learning by making these design mistakes 😕

Reply:
@clason Do you have any rule of thumb on when to use fields *vs* when to use aliasing ? Is aliasing to a virtual node name (only existing through aliasing) bad practice, or is it ok ?